File: ios/App/App/plugins/Voip/VoipPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

CAP_PLUGIN(VoipPlugin, "VoipPlugin",
  CAP_PLUGIN_METHOD(registerUser, CAPPluginReturnPromise);
  CAP_PLUGIN_METHOD(getCurrentUser, CAPPluginReturnPromise);
  CAP_PLUGIN_METHOD(startCall, CAPPluginReturnPromise);
  CAP_PLUGIN_METHOD(endCall, CAPPluginReturnPromise);
)

File: ios/App/App/plugins/Voip/VoipPlugin.swift
import Foundation
import Capacitor

@objc(VoipPlugin)
public class VoipPlugin: CAPPlugin {
    
  private let implementation = Voip()

  @objc public func registerUser(_ call: CAPPluginCall) {
    call.resolve()
  }

  @objc public func getCurrentUser(_ call: CAPPluginCall) {
    let currentUser = implementation.getCurrentUser();
    call.resolve(["user": currentUser])
  }

  @objc public func startCall(_ call: CAPPluginCall) {
    call.resolve();
  }
    
  @objc public func endCall(_ call: CAPPluginCall) {
      call.resolve();
  }
}
